pkg/skaffold/runner/v1/deploy.go

```diff
@@ -53,7 +52,6 @@ func (r *SkaffoldRunner) DeployAndLog(ctx context.Context, out io.Writer, artifa
 	logger.SetSince(time.Now())
 	// First deploy
 	if err := r.Deploy(ctx, out, artifacts); err != nil {
-		eventV2.TaskFailed(constants.Deploy, err)
```

Comment:
Wouldn't this cause the events to be not sent?

Reply:
It shouldn't as the `runner.Deploy` function has these same function calls. I originally made a mistake as I thought that `runner.DeployAndLog` and `runner.Deploy` were completely separate, but realized that `runner.Deploy` always gets called, either independently or within `runner.DeployAndLog`. That's why I made the original PR, and this is just redoing that PR